3 - PB/Documentazione esterna/Specifica Tecnica/Specifica Tecnica.typ

```diff
@@ -146,7 +147,227 @@ Ciò avviene prestando attenzione alle tecnologie impiegate sia nel front-end ch
+
+  \
+  Per ogni attributo è presente il corrispondente metodo get.
+  Sono inoltre presenti i metodi:
+  - *getLevels*: ritorna una lista contenente le liste di bin che rappresentano i livelli della zona;
+  - *getColumns*: ritorna una lista contenente le liste di bin che rappresentano le colonne della zona;
+  - *getMaxUsedLevel*: ritorna il numero dell'ultimo livello della zona con almeno un bin contenente un prodotto;
+  - *getMaxUsedColumn*: ritorna il numero dell'ultima colonna della zona con almeno un bin contenente un prodotto.
+
+\
+- *Product*:
+
+  rappresenta un prodotto, i suoi attributi sono:
+  - *id*: interno che rappresenta il codice identificativo univoco del prodotto;
+  - *name*: stringa che rappresenta il nome del prodotto;
+  - *weight*: numero in virgola mobile che rappresenta il peso del prodotto;
+  - *height*: numero in virgola mobile che rappresenta l'altezza del prodotto;
+  - *length*: numero in virgola mobile che rappresenta la profondità del prodotto;
+  - *width*: numero in virgola mobile che rappresenta la larghezza del prodotto;
+  - *categories*: lista di stringhe che rappresentano le categorie del prodotto.
+
+  Per ogni attributo è presente il corrispondente metodo get.
+
+\
+- *Order*:
+
+  rappresenta la richiesta di uno spostamento di un prodotto tra due bin.
+  I suoi attributi sono:
+  - *id*: interno che rappresenta il codice identificativo univoco della richiesta;
+  - *startPoint*: riferimento al bin iniziale;
+  - *endPoint*: riferimento al bin finale;
+  - *product*: riferimento al prodotto da spostare.
+
+  Per ogni attributo è presente il corrispondente metodo get.
+
+\
+- *Floor*:
+
+  rappresenta il piano dell'ambiente 3D, i suoi attributi sono:
+  - *length*: numero in virgola mobile che rappresenta la profondità del piano;
+  - *width*: numero in virgola mobile che rappresenta la larghezza del piano;
+  - *SVG*: stringa che contiene il path al file SVG se presente. Nel caso in cui non sia stato identificato nessun file SVG, la variabile è `null`.
+
+  Per ogni attributo è presente il corrispondente metodo get.
+
+  Potendo generare l'oggetto `Floor` con modalità diverse a seconda della presenza del file SVG, la sua creazione è gestita tramite il design pattern Strategy e le relative classi:
+
+  - l'interfaccia `FloorStrategy`: rappresenta la creazione di un elemento `Floor`;
+  - `StandardFloorStrategy`: implementa `FloorStrategy` e rappresenta la creazione di un elemento `Floor` senza file SVG;
+  - `CustomFloorStrategy`: implementa `FloorStrategy` e rappresenta la creazione di un elemento `Floor` con file SVG;
+  - `FloorStrategyContext`: utilizza un `FloorStrategy` per generare un elemento `Floor`.
+
+#figure(
+  image("./imgs/Business Layer classes.png", width: 100%),
+  caption: [
+    Diagramma delle classi del layer di business
+  ],
+)
+
+#figure(
+  image("./imgs/Business Layer floor.png", width: 100%),
```

Comment:
Composizione errata. 
![image](https://github.com/Error-418-SWE/Documenti/assets/77143275/5cf87163-8666-4b18-9e31-258bd2e100d4)

Reply:
quindi rispetto a come ho fatto cosa devo cambiare? la composizione con aggregazione o altro?

Reply:
per i diagrammi possiamo discuterne in meeting di retrospettiva